File: 0372-super-pow/0372-super-pow.py
class Solution:
    def superPow(self, a: int, b: List[int]) -> int:
        mod = 1337
        def powmod(x, y):
            result = 1
            x %= mod
            for _ in range(y):
                result = (result * x) % mod
            return result
         
        ans = 1
        for digit in b:
            ans = powmod(ans, 10) * powmod(a, digit) % mod
        return ans

File: 0372-super-pow/0372-super-pow.java
class Solution {
    private static final int mod = 1337;

    public int superPow(int a, int[] b) {
        int result = 1;
        for (int digits : b){
            result  = powMod(result , 10) * powMod(a, digits) % mod;

        }
        return result;
    }
    public int powMod(int x, int y){
        int res = 1;
        x %= mod;
        for(int  i = 0; i < y; i++){
            res = ( res * x) % mod;
        }
        return res;

    }
}